todo-app/src/App.js

```diff
@@ -1,13 +1,83 @@
-import React, { Component } from 'react';
+//jshint esnext: true
+
+import React from 'react';
+
+import TodoList from './Components/todoList';
+import AddTodoItem from './Components/AddTodoItem';
+import './App.css';
+
+
+export default class App extends React.Component {
+
+  
+  state= {
+    todos: [],
+    isEditing: null
+  }
+
+
+  handleToggleEditing = (id) => {
+    this.setState({ isEditing: id || null })
+  }
+
+  handleAddTodo = (text, deadline) => {
+    
+    const newItem = {
+        id: this.state.todos.length + 1 || 1,
```

Comment:
We very briefly mentioned this on Sunday. What if I create three todos from scratch, so that I have a situation like this:

    [
      { title: "Thing 1", id: 1 },
      { title: "Thing 2", id: 2 },
      { title: "Thing 3", id: 3 },
    ]

Now imagine that I delete "Thing 2", and try to add a new todo. What will happen? You can start up your own app here and try it, you'll see the error in the dev console.